src/python/setup.py.in

```diff
@@ -42,14 +56,26 @@ for module in modules:
         runtime_library_dirs=runtime_library_dirs,
         cython_directives = {'language_level': str(sys.version_info[0])},))
 
+ext_modules = cythonize(ext_modules)
+
+ext_modules.append(Extension(
+    'gudhi.hera',
+    sources = [source_dir + 'hera.cc'],
+    language = 'c++',
+    include_dirs = include_dirs +
+        ['@HERA_WASSERSTEIN_INCLUDE_DIR@',
+        get_pybind_include(False), get_pybind_include(True)],
+    extra_compile_args=extra_compile_args + [@GUDHI_PYBIND11_EXTRA_COMPILE_ARGS@],
+    ))
+
 setup(
     name = 'gudhi',
     packages=["gudhi",],
     author='GUDHI Editorial Board',
     author_email='gudhi-contact@lists.gforge.inria.fr',
     version='@GUDHI_VERSION@',
     url='http://gudhi.gforge.inria.fr/',
-    ext_modules = cythonize(ext_modules),
-    install_requires = ['cython','numpy >= 1.9',],
-    setup_requires = ['numpy >= 1.9',],
+    ext_modules = ext_modules,
+    install_requires = ['cython','numpy >= 1.9','pybind11',],
+    setup_requires = ['numpy >= 1.9','pybind11',],
```

Comment:
To be discussed, but maybe we could make it optional if submodule is updated and init and if pybind11 is installed.

Reply:
Normally this was written so that if you type `pip3 install .` instead of `python3 setup.py build_ext --inplace`, it automatically installs pybind11 before building gudhi. Sadly, I seem to have broken the lazy get_pybind_include trick, because it fails the build, and you need to execute it a second time :disappointed: 
I think we should make pybind11 mandatory, like cython. We are going to have more pieces depending on it, it is an easy dependency available everywhere (pip, conda, linux distributions).
Several python dependencies are only runtime dependencies (you only need them to run the code, not when building the package) as long as you don't build the doc or run the tests, so it makes sense not to require them at build time. CGAL is optional mostly because of the license in my opinion. But cython and pybind11 are really necessary to build the package (not to use it though).
Missing the submodule: people using tarballs will always have it. People badly checking out gudhi but not its submodules, I would consider a bug. It doesn't seem like a case worth supporting.